src/base/editor-context.ts

```diff
@@ -27,12 +29,26 @@ export interface EditorContext {
     readonly args?: Args;
 }
 
+export interface EditModeListener {
+    editModeChanged(newValue: string, oldvalue: string): void;
+}
 @injectable()
-export class EditorContextService {
+export class EditorContextService implements IActionHandler {
 
     @inject(GLSP_TYPES.SelectionService) protected selectionService: SelectionService;
     @inject(MousePositionTracker) protected mousePositionTracker: MousePositionTracker;
     @inject(TYPES.ModelSourceProvider) protected modelSource: () => Promise<ModelSource>;
+    editMode: string;
+
+    constructor(@multiInject(GLSP_TYPES.SelectionListener) @optional() protected editModeListeners: EditModeListener[] = []) { }
```

Comment:
I don't really understand why we inject `@multiInject(GLSP_TYPES.SelectionListener)` here and call them `editModeListeners`. Do you mean to inject `IEditModeListener`?

Reply:
Good catch. Yes this should be IEditModeListener